### DISCUSSION TITLE
Why does parcel expect people to use CDN?

### DISCUSSION THREAD

**Discussion by freehuntx** [parcel-bundler/parcel#5099] [General]
Everywhere where people ask or complain about the hashes in file names i read the excuse its because of CDN.
And if you want to turn it off because you simply dont need it (for example because you use express) you have to write some Namer plugin and blow up your configs.

Merge requests adding a option to turn this behaviour off get denied.

I dont understand this!
It really frustrates me because on one side parcel has the potential to simplify project setups but on the other side it add such complexity.

Does parcel expect the users to use CDN? 

Edit:
An workaround is to not use script tags to include scripts, but use something like this and compile the js file seperately.
Because then parcel does not add hashes to js files...
index.html:
```
<!DOCTYPE html>
<html>
  <head></head>
  <body>
    <script>
      const script = document.createElement('script')
      script.src = './client.js'
      document.body.appendChild(script)
    </script>
  </body>
</html>
```

**Comment by mischnic** [comment]
Another reason why we add hashes is having unique filenames.

And it's not "CDN" but "Caching": https://hacks.mozilla.org/2017/01/using-immutable-caching-to-speed-up-the-web/

> Merge requests adding a option to turn this behaviour off get denied.

?

**Comment by freehuntx** [comment]
The issue with "unique filenames" is your dist folder blows up unless you add another plugin/line to delete it always.
I get the point this is maybe usefull in some production builds. But why enforce it? Why not give people the freedom to turn this off?
You also dont enforce minifying. And thats also mainly used to "speed up the web".
But for some reason since parcel 1 this behaviour is enforced and i dont understand why.

And to answer your "?":
https://github.com/parcel-bundler/parcel/pull/3698
https://github.com/parcel-bundler/parcel/pull/2295
https://github.com/parcel-bundler/parcel/pull/2200
https://github.com/parcel-bundler/parcel/pull/1934

The list is even longer.

**Comment by freehuntx** [comment]
Once in a while i come back to parcel to see if it added this simple feature. And after many years i still dont see improvement.
Its really sad.
